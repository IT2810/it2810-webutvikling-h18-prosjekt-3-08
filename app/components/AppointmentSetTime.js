import React from 'react';
import {StyleSheet, Text, View, Button, TouchableOpacity, Keyboard} from 'react-native';
import DateTimePicker from 'react-native-modal-datetime-picker';
import moment from 'moment'; // For å håndtere/formatere valgt tid

export default class AppointmentSetTime extends React.Component {

    constructor(props) {
        super(props)
        this.state = {
            isVisible: false,
            chosenTime: 'xx:xx',
            picked: false
        }
    }

    handlePicker = (time) => {
        Keyboard.dismiss()
        let formatedTime = moment(time).format('HH:mm')
        this.setState({
            isVisible: false,
            chosenTime: formatedTime,
            picked: true
        });
        this.props.onSelectTime(formatedTime)
    };

    showPicker = () => {
        this.setState({
            isVisible: true
        })
    };

    hidePicker = () => {
        this.setState({
            isVisible: false
        })
    };

    render() {
        let textStyle = this.state.picked ? styles.text_chosen : styles.text
        return (
            <View style={styles.container}>
                <TouchableOpacity  onPress={this.showPicker}>
                    <Text style={textStyle}>{this.state.chosenTime}</Text>
                </TouchableOpacity>
                <DateTimePicker 
                        time={this.state.chosenTime} 
                        isVisible={this.state.isVisible}
                        onConfirm={this.handlePicker} 
                        onCancel={this.hidePicker} 
                        mode={"time"}
                        is24Hour={true}
                    />
            </View>);
    }
}


const styles = StyleSheet.create({
    container: {
        backgroundColor: '#fff',
        alignItems: 'center',
        padding: 10,
        paddingTop: 0
    },
    text: {
        fontSize: 24,
        textDecorationLine: 'underline',
        textDecorationColor: '#ededed'
    },
    text_chosen: {
        fontSize: 24,
        textDecorationLine: 'underline',
        textDecorationColor: '#A9E2F3'
    }
});