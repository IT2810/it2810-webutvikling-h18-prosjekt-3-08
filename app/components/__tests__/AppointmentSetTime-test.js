import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import AppointmentSetTime from '../AppointmentSetTime';


it("Renders correctly", () => {
    const tree = renderer.create(<AppointmentSetTime/>).toJSON();
    expect(tree).toMatchSnapshot();
});
