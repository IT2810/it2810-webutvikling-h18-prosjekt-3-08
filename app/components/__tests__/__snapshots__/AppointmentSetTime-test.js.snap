// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "padding": 10,
      "paddingTop": 0,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 24,
          "textDecorationColor": "#ededed",
          "textDecorationLine": "underline",
        }
      }
    >
      xx:xx
    </Text>
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onRequestClose={[Function]}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "black",
          "bottom": 0,
          "height": 1334,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": 750,
        }
      }
    />
    <View
      hideModalContentWhileAnimating={false}
      onModalHide={[Function]}
      pointerEvents="box-none"
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
          "margin": 10,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        }
      }
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 13,
              "marginBottom": 8,
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "borderBottomColor": "#d5d5d5",
              "borderBottomWidth": 0.5,
              "padding": 14,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#8f8f8f",
                  "fontSize": 13,
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Pick a date
          </Text>
        </View>
        <View
          onStartShouldSetResponderCapture={[Function]}
        >
          <View>
            <RCTDatePicker
              date={1539964665708}
              minuteInterval={1}
              mode="time"
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": 216,
                }
              }
            />
          </View>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "#d5d5d5",
              "borderTopWidth": 0.5,
              "height": 57,
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#007ff9",
                  "fontSize": 20,
                  "fontWeight": "normal",
                  "textAlign": "center",
                },
                undefined,
              ]
            }
          >
            Confirm
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 13,
            "height": 57,
            "justifyContent": "center",
            "marginBottom": 0,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#007ff9",
                "fontSize": 20,
                "fontWeight": "600",
                "padding": 10,
                "textAlign": "center",
              },
              undefined,
            ]
          }
        >
          Cancel
        </Text>
      </View>
    </View>
  </Modal>
</View>
`;
